TROUBLESHOOT.md

```diff
@@ -0,0 +1,165 @@
+to deploy the reporter most conveniently as per user’s need.
+
+## General Debugging
+### Records are not being written to BigQuery
+With the help of metrics available as a part of 0.4.0 release of the connector. 
+Users should be able to track the number of records that enter the sink(writer) and the 
+number of records successfully written to BigQuery. 
+If records are not being written to BigQuery, then records are stuck in either of the two phases:
+#### The records are not arriving at the sink
+- The problem lies with the pipeline, the previous chain of subtasks that are performed before 
+sink is called.
+- The pipeline is not processing the passing the records forward for the sink to serialize and 
+write to BigQuery.
+#### The records are arriving at the sink but not being successfully written to BigQuery.
+- Check the logs or error message for the following errors:
+     - #### `BigQuerySerializationException`
+     - This message illustrates that the record(s) could not be serialized by the connector. 
+     - The error message would also contain the actual cause for the same.
+     - #### `BigQueryConnectorException`
+     - Refer to Java Docs for the BigQuery Storage Write API proper usage and the errors that might
+arise from the same.
+- Ensure checkpointing is enabled and properly configured. Setting too large checkpointing 
+intervals can cause records to not be written for long periods of time.
+
+## Known Issues/Limitations
+<b><i>Note: Users must go through the readme documentation for the connector first to 
+ensure proper usage of the connector, the connector also has certain limitations which are 
+documented in the readme.</i></b>
+### Flink Application with Windows
+- Windows are tricky to deal with in Flink. Flink relies on watermarks to close windows. 
+- In case no records are written to BigQuery, users can observe the flow of records through 
+the application to ensure the sink (writer) is receiving records. 
+- In case the sink is not receiving any records, then the windows are not being closed. 
+- The incoming records are windowed together by Flink Application that is continuously waiting 
+for closing events (as per the windowing condition) to arrive.
+- [Flink's Documentation on Debugging Windows](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/ops/debugging/debugging_event_time/)
+
+### BigQuery sink does not appear to be writing consistently
+- The Connector in At Least Once mode writes to BigQuery at two instances - when the append request 
+reaches its limit and when checkpoints are triggered. 
+So, the write to BigQuery would not be continuous; it would occur in stages where each write event 
+would coincide with the checkpoint time.
+- The write to BigQuery API would not be at a steady rate, it would contain spiking intervals 
+pertaining to each checkpoint event when data is being written to BigQuery.
```

Comment:
Lets mention that this is true for exactly-once mode

Reply:
Done.